src/client.rs

```diff
@@ -97,47 +101,94 @@ impl ActivityPubClient {
     }
 
     pub async fn follow_actor(&self, actor_uri: &str) -> Result<()> {
-        let Actor { id, inbox, .. } = self.get_actor(actor_uri).await?;
         let base = &self.base;
-        info!(%id, %inbox, "sending follow request to inbox");
+        let actor: Actor = self.get_actor(actor_uri).await?;
+        let actor_id = actor.id.as_ref().ok_or(Error::StatusAndMessage {
+            status: StatusCode::BAD_REQUEST,
+            message: "actor has no id",
+        })?;
+        let id = actor_id
+            .parse::<http::Uri>()
+            .map_err(|_e| Error::InvalidUri {
+                uri: actor_id.clone(),
+            })?;
+        info!(
+            "sending follow request to inbox: {:?} {:?}",
+            id, actor.inbox
+        );
```

Comment:
`tracing` supports including values as named key/value pairs in the JSON output as opposed to interpolating them into the message itself:
```rust
info!(?id, inbox=?actor.inbox, "seding follow request to inbox");
```